velox/functions/sparksql/specialforms/GetStructField.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/sparksql/specialforms/GetStructField.h"
+#include "expression/Expr.h"
+#include "vector/ComplexVector.h"
+#include "vector/ConstantVector.h"
+#include "velox/expression/PeeledEncoding.h"
+
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::functions::sparksql {
+
+void GetStructFieldExpr::evalSpecialForm(
+    const SelectivityVector& rows,
+    EvalCtx& context,
+    VectorPtr& result) {
+  VectorPtr input;
+  VectorPtr ordinalVector;
+  inputs_[0]->eval(rows, context, input);
+  inputs_[1]->eval(rows, context, ordinalVector);
+  VELOX_USER_CHECK(ordinalVector->isConstantEncoding());
```

Comment:
I suppose we can move the checks inside `constructSpecialForm` and extract the constant value there. Some example in decimal round:

https://github.com/facebookincubator/velox/blob/04bfdffcf4f528f33e03fb3a74e103ffab36f222/velox/functions/sparksql/specialforms/DecimalRound.cpp#L229-L245

Reply:
update